operations/research-and-development/engineering/plugin-release-process.md

```diff
@@ -0,0 +1,146 @@
+
+The next steps are to add the plugin to the Plugin Marketplace. The instructions are included in the return message upon a successful `cutplugin` command.
+
+### Bundle a plugin release version to a Mattermost server release
+
+Plugins that are released with Mattermost are called bundled plugins. These plugins are included with the software and need only to be configured.
+
+* `git pull` the latest master branch on mattermost-server
+* Create a new branch so you can modify the plugin versions
+  * `git checkout -b bundle-plugins-v5.20`
+  * Use branch naming convention `bundle-plugins-vX.XX`
+* Edit `Makefile`
+  * Locate `# Plugins Packages` comment
+  * Modify plugin release versions
+* Create PR against master branch with following:
+  * **Title:** `Update bundled plugins for vX.XX`
+  * **Summary:** List of updated plugins
+    * Ideally includes from version -> to version for each plugin
+
+### Publish a plugin release version to the Plugin Marketplace
+
+The steps to have a plugin version added the Plugin Marketplace are included with the success of an `/mb cutplugin` slash command.
+
+### Release to `community.mattermost.com`
+
+Once the Marketplace PR is merged, request an update on the plugin in the `~community-configuration`[https://community.mattermost.com/core/channels/community-configuration] channel.
+
+## Security upgrade process
+
+(**`TODO`**: Automate checking all released plugins through CLI, cron, or GH webhook event)  
+(**`TODO`**: User must be repo admin to see and resolve automated security issue)  
+(**`TODO`**: Need method to hook to tell us when security issue is found)  
+(**`TODO`**: PR for security updates should be discrete)  
+(**`TODO`**: investigate `npm ls hoek`)
+
+Security alerts are displayed when viewing a GitHub repo and are resolved via the automated `dependabot` tool
+
+### Updating npm Dependencies CLI
+
+* `git checkout latest master`
+* `git checkout -b bump-dependency-versions`
+* `cd webapp/`
+  * `npm-check -E -u` to view the changes interactively
+    `npm-check -E -y` to update without interactive
+* `git add package-lock.json package-lock.json`
+* `git commit -m "Update dependencies"`
+* `git push --set-upstream origin bump-dependency-versions`
+* Create PR
+  * Title: `Update Dependencies` (Will automatically get set)
+  * Summary: Update dependencies
+
+### Updating Security Alerts Through CLI
+
+* `git checkout latest master`
+* `git checkout -b npm-audit-fix`
+* `cd webapp/`
+* `npm audit` - will return list of security issues
+* `npm audit fix` - updates ``package-lock.json`` dependencies
+* `git add package-lock.json`
+* `git commit -m "Update dependencies"`
+* `git push --set-upstream origin npm-audit-fix`
+* Create PR
+  * Title: `Update Dependencies` (Will automatically get set)
+  * Summary: &lt;library\_name&gt; &lt;from\_ver&gt; -&gt; &lt;to\_ver&gt;
```

Comment:
Is the inclusion of the "&lt;" and "&gt;" correct here?

Reply:
Ah, I see how it's meant to render - should it be:

<library\_name>; <from\_ver> -> <to\_ver>

Reply:
Updated :+1: